src/repeated_game.jl

```diff
@@ -0,0 +1,364 @@
+"""
+function initialize_LP_matrices(rpd::RepGame2, H)
+    # Need total number of subgradients
+    nH = size(H, 1)
+
+    # Create the c vector (objective)
+    c = zeros(2)
+
+    # Create the A matrix (constraints)
+    A_H = H
+    A_IC_1 = zeros(1, 2)
+    A_IC_2 = zeros(1, 2)
+    A_IC_1[1, 1] = -rpd.delta
+    A_IC_2[1, 2] = -rpd.delta
+    A = vcat(A_H, A_IC_1, A_IC_2)
+
+    # Create the b vector (constraints)
+    b = Array(Float64, nH + 2)
+
+    return c, A, b
+end
+
+"""
+Given a constraint w ∈ W, this finds the worst possible payoff for agent i
+
+The output of this function is used to create the values associated with
+incentive compatibility constraints
+
+The arguments for this function are
+  * rpd: Two player repeated game
+  * H: Subgradients used to approximate value set
+  * C: Hyperplane levels for value set approximation
+  * i: Which player want worst value for
+"""
+function worst_value_i(rpd::RepGame2, H::Array{Float64, 2}, C::Array{Float64, 1}, i::Int)
+    # Objective depends on which player we are minimizing
+    c = zeros(2)
+    c[i] = 1.0
+
+    # Lower and upper bounds for w
+    lb = [-Inf, -Inf]
+    ub = [Inf, Inf]
+
+    lpout = linprog(c, H, '<', C, lb, ub, ClpSolver())
+    if lpout.status == :Optimal
+        out = lpout.sol[i]
+    else
+        out = minimum(rpd.sg.players[i].payoff_array)
+    end
+
+    return out
+end
+
+"See worst_value_i for documentation"
+worst_value_1(rpd::RepGame2, H::Array{Float64, 2}, C::Array{Float64, 1}) =
+    worst_value_i(rpd, H, C, 1)
+"See worst_value_i for documentation"
+worst_value_2(rpd::RepGame2, H::Array{Float64, 2}, C::Array{Float64, 1}) =
+    worst_value_i(rpd, H, C, 2)
+"See worst_value_i for documentation"
+worst_values(rpd::RepGame2, H::Array{Float64, 2}, C::Array{Float64, 1}) =
+    (worst_value_1(rpd, H, C), worst_value_2(rpd, H, C))
+
+#
+# Outer Hyper Plane Approximation
+#
+"""
+Approximates the set of equilibrium value set for a repeated game with the
+outer hyperplane approximation described by Judd, Yeltekin, Conklin 2002
+
+NOTE: If your code fails then it might be the case that the value set is
+      only the value which corresponds to the pure action nash equilibrium
+      or you might just need more points to be precise enough for the
+      Polyhedra library to be able to convert to the vertice representation.
+
+The arguments are
+  * rpd: 2 player repeated game
+
+The keyword arguments are
+  * nH: Number of subgradients used in approximation
+  * tol: Tolerance in differences of set
+  * maxiter: Maximum number of iterations
+  * verbose: Whether to display updates about iterations and distance
+  * nskipprint: Number of iterations between printing information (verbose=true)
+  * pane_check: Whether to perform a check about whether a pure action ne exists
+
+"""
+function outerapproximation(rpd::RepGame2; nH=32, tol=1e-8, maxiter=500,
+                            verbose=false, nskipprint=50)
```

Comment:
Thanks for the comment (and for putting together `Polyhedra`!) @blegat.

If I understand correctly, your suggestion is to implement a function that chooses which backend to use for `Polyhedra`? I'm currently forcing users to use CDDLibrary (which is probably the backend most should use for now anyways), but it would definitely be nice to give some options.

I hadn't seen MPB's default code before. That looks slick, but I'll need a few minutes to make sense of it.